File: go/tripservice_original.go
package trip

import (
	"github.com/pkg/errors"
	"github.com/sandromancuso/trip-service-kata/go/user"
	"github.com/sandromancuso/trip-service-kata/go/trip"
)

type Service struct {
	tripDAO *trip.Dao
}

// New ...
func NewTripService(tripDao *trip.Dao) *Service {
	return &Service{
		tripDAO: tripDao,
	}
}

func (this *Service) getTripByUser(friend *user.User) ([]trip.Trip, error) {

	var trips []trip.Trip

	friends, err := friend.Friends()
	if err != nil {
		return trips, err
	}
	loggedUser, err := user.CurrentSession().GetLoggedUser()
	if err != nil {
		return trips, err
	}

	var isFriend bool
	if loggedUser != nil {
		for _, friend := range friends {
			if *loggedUser == friend {
				isFriend = true
				break
			}
		}
		if isFriend {
			return this.tripDAO.FindTripByUser(friend)
		}
		return trips, err
	} else {
		return trips, errors.New("user not logged in")
	}
}

File: python-objects/TripService_Original.py
from TripDAO import TripDAO
from UserNotLoggedInException import UserNotLoggedInException
from UserSession import UserSession


class TripService(object):
    def get_trips_by_user(self, user):
        logged_user = UserSession.get_instance().get_logged_user()
        is_friend = False
        trip_list = []
        if logged_user:
            for friend in user.get_friends():
                if friend is logged_user:
                    is_friend = True
                    break
            if is_friend:
                trip_list = TripDAO.find_trips_by_user(user)
            return trip_list
        else:
            raise UserNotLoggedInException()
